### DISCUSSION TITLE
Running TinyUSB on my own PCB with SAMD21

### DISCUSSION THREAD

**Discussion by 2400audio** [hathach/tinyusb#1069] [Q&A]
Hi 
I'm trying to make a USB MIDI interface using the SAMD21. 
I can run the MIDI example on my SAMD21 xplained pro board (SAMD21J18A) after implementing it into a simple Atmel start project, and this is works. Great!
However, I have now made my own PCB with an SAMD21G17D MCU (smaller version), but I simply don't know how to change TinyUSB library to work on my own board instead of the xplained pro board. I have tried to modify the board.h file without success, and I'm not sure that's the right approach anyway. 
Any hints?

Additional question:
Does tinyUSB set up the needed clock sources on the MCU?  

Niels:-)

**Comment by hathach** [comment]
the tinyusb library does not, but the board is set up by example via bsp. you can take a look at samd21 bsp code for reference https://github.com/hathach/tinyusb/tree/master/hw/bsp/samd21

**Comment by mws-rmain** [comment]
I'm doing something very similar but in reverse. Initial development on a SAMD21G17D (curiousity nano), now moved to a SAMD21J18A (SAMD21-XPlained), eventually to my own design using a SAMD21J17D.
I've integrated the TinyUSB stack into an Atmel(/Microchip) Studio project, and have had no problem switching processors within Studio (the 'change device' button on the 'device' tab of project properties dialog).
It has been two months since your original post. Have you resolved this, or are you still having difficulties?

**Reply by 2400audio** [reply]
Thanks for replying to this. I did manage to get it working. The problem was, that I did not configure the clock settings correctly in my Atmel start project. When I have 48Mhz running on Generic clock generator 7, it it working.
